Description: The `cupsGetNamedDest` function did not use the local default printer
 During some refactoring done to cupsGetNamedDest function used in many
 different parts of the code a regression has been introduced causing the
 per-user default printer setting (in ~/.cups/lpoptions) to be ignored.
Author: Michael Sweet <michael.r.sweet@gmail.com>
Origin: upstream, https://github.com/apple/cups/commit/c536b6c583abd9ea1b750f15e887b313ed7ad951
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1729910

Index: cups-2.2.4/cups/dest.c
===================================================================
--- cups-2.2.4.orig/cups/dest.c
+++ cups-2.2.4/cups/dest.c
@@ -2244,7 +2244,7 @@ cupsGetNamedDest(http_t     *http,	/* I
       dest_name = cups_get_default(filename, defname, sizeof(defname), &instance);
     }
 
-    if (!name)
+    if (!dest_name)
     {
      /*
       * Still not there?  Try the system lpoptions file...
Index: cups-2.2.4/cups/testdest.c
===================================================================
--- cups-2.2.4.orig/cups/testdest.c
+++ cups-2.2.4/cups/testdest.c
@@ -105,6 +105,14 @@ main(int  argc,				/* I - Number of comm
   }
   else if (!strncmp(argv[1], "ipp://", 6) || !strncmp(argv[1], "ipps://", 7))
     dest = cupsGetDestWithURI(NULL, argv[1]);
+  else if (!strcmp(argv[1], "default"))
+  {
+    dest = cupsGetNamedDest(CUPS_HTTP_DEFAULT, NULL, NULL);
+    if (dest && dest->instance)
+      printf("default is \"%s/%s\".\n", dest->name, dest->instance);
+    else
+      printf("default is \"%s\".\n", dest->name);
+  }
   else
     dest = cupsGetNamedDest(CUPS_HTTP_DEFAULT, argv[1], NULL);
 
